lessons/day_07/day_07.md

```diff
@@ -8,23 +8,366 @@
+    - unrestricted arbitrary content
+    - since **adding other entries to the error map are highly discouraged** —but not prohibited— this might be the RIGHT PLACE for engineers to add extra info whenever required
+- `data` and `errors` entries can **coexist** in the response
+
+### 3. One Object to bring them all — `data` & `error` are siblings
+
+You may have both a `data` and an `error` properties in the Response body, containing expected data as well as 1 or more errors.
+
+Going back to our characters list example, let's say the `name` property is **nullable**  for `Istary` but **non-nullable** for `Hobbit`  and one of the records has a null value; this is what we get as a response:
+
+**TypeDef**
+
+```graphql
+interface Character {
+  id: ID
+  name: String
+  ....
+}
+
+type Hobbit implements Character {
+  id: ID
+  name: String!
+  ....
+}
+
+type Istari implements Character{
+  id: ID
+  name: String
+}
+```
+
+**Query**
+
+```graphql
+query Characters{
+  characters {
+    name
+  }
+}
+```
+
+**Response**
+
+```json5
+{
+  "errors": [
+    {
+      "message": "Cannot return null for non-nullable field Hobbit.name.",
+      "locations": [
+        {
+          "line": 3,
+          "column": 5
+        }
+      ],
+      "path": [
+        "characters", // the query operation name
+        0, // the index of the record related to the list of characters returned
+        "name" // the field name
+      ],
+      "extensions": {
+        "code": "INTERNAL_SERVER_ERROR",
+        "exception": {
+          "stacktrace": [.....] // a lot of stuff here
+        }
+      }
+    }
+  ],
+  "data": {
+    "characters": [
+      null, // <-- there you go, the first records is null due to the error but the rest can be delivered
+      { "name": "Sam" },
+      { "name": "Meriadoc" },
+      { "name": "Peregrin" },
+      { "name": "Arwen" },
+      { "name": "Elrond" },
+      { "name": "Celebrían" },
+      { "name": "Drogo" },
+      { "name": "Saradoc" },
+      { "name": "Esmeralda" },
+      { "name": "Gandalf" },
+      { "name": null }, // this Istari has now a null name, but it's fine for GraphQL because it's nullable
+      { "name": "Radagast" },
+      { "name": "Galadriel" }
+    ]
+  }
+}
+```
+
+### 4. and in your mind bind them — **"errors"** as **"unrequested results"**
+
+Let's stay put for a moment and analyse what just happened above.
+
+1. One of our records doesn't respect the contract for it's type
+2. An error is added to the response describing that particular fault
+3. The execution could be performed for the rest of the records and they were added to the `data` entry as expected
+
+So:
+
+1. Should we consider the whole as an error?
+2. Should we tell the user something about the fault? what? how? when?
+3. Can/Should the user do anything about it?
+
+... and many other questions can derive from this behavior!  
+
+So what do we do about this?  
+There's no "the right answer" for that.  
```

Comment:
Would it be better without the "the"? "_There´s no 'right answer' for that_"

Reply:
it refers to the sentence "The right answer" as a superlative assertion. There might be "a right answer" taking in consideration the context, but it'll never be "THE" right answer.

Reply:
Yes, i know the sentence, but given the context i think just "right answer" will suffice. it is up to you 😉